cmd/client/main.go

```diff
@@ -2,79 +2,235 @@ package main
 
 import (
 	"bytes"
+	"context"
 	"crypto/rand"
+	"errors"
 	"fmt"
 	"io"
 	"net"
 	"os"
+	"os/signal"
+	"runtime"
 	"sync/atomic"
 	"time"
 
 	flag "github.com/spf13/pflag"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/cilium/test-connection-disruption/internal"
 )
 
-const MSG_SIZE = 256 // should be synced with the server
+const maxAttempts = 30
 
-var DispatchInterval time.Duration
+func init() {
+	internal.ErrExit("being nice", internal.BeNice())
```

Comment:
Wouldn't it make more sense to limit tcd pods resources instead?

Reply:
I don't think so, since 1 vCPU is really not equivalent across all platforms where this is run, for example comparing a non-virtualized core on a desktop kind cluster and a core in an LVH VM on an oversubscribed CI instance.

The required resources of the server pod(s) also depends on how many clients there are. It also wouldn't completely solve the problem we're seeing in Cilium CI (conn-disrupt stealing CPU time from clang on agent restart), since both workloads would still have the same priority.